apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -176,109 +176,123 @@ oracle_ttl_t() ->
-                                    ArgumentTypes = [word, word, sign_t(), RType],
-                                    [_, _, Sign0, R] = get_args(ArgumentTypes, Data),
-                                    Callback = fun(API, ChainState) ->
-                                        API:oracle_respond(OracleKey, QueryKey, to_sign(Sign0), R, DeltaRTTL, ChainState) end,
-                                    cast_chain(Callback, State);
-                                {error, _} = Err -> Err
-                            end
-                        end,
-                    DynCost = state_gas_cost(oracle_response, DeltaRTTL),
-                    ?TEST_LOG("~s computed gas cost ~p from relative TTL ~p", [?FUNCTION_NAME, DynCost, DeltaRTTL]),
-                    {ok, DynCost, Callback2}
+                    case API:oracle_response_format(OracleKey, State) of
+                        {error, _} = Err -> Err;
+                        {ok, RFormat} ->
+                            ArgumentTypes = [word, word, sign_t(), RFormat],
+                            [_, _, Sign0, R] = get_args(ArgumentTypes, Data),
+                            {ok, Tx} = API:oracle_respond_tx(OracleKey, QueryKey, R, DeltaRTTL, State),
+                            SizeGas = size_gas(Tx),
+                            StateGas = state_gas(oracle_response, DeltaRTTL),
+                            DynGas = SizeGas + StateGas,
+                            ?TEST_LOG("~s computed gas ~p - size gas: ~p, state gas: ~p (relative TTL: ~p)",
+                                      [?FUNCTION_NAME, DynGas, SizeGas, StateGas, DeltaRTTL]),
+                            Callback = fun(ChainAPI, ChainState) ->
+                                ChainAPI:oracle_respond(Tx, to_sign(Sign0), ChainState) end,
+                            {ok, DynGas, fun() -> cast_chain(Callback, Chain) end}
+                    end
             end
     end.
 
-oracle_call_extend(_Value, Data, State) ->
+oracle_call_extend(_Value, Data, #chain{api = API, state = State} = Chain) ->
     ArgumentTypes = [word, sign_t(), oracle_ttl_t()],
     [Oracle, Sign0, TTL] = get_args(ArgumentTypes, Data),
-    case chain_ttl_delta(TTL, State) of
+    case chain_ttl_delta(TTL, Chain) of
         {error, _} = Err -> Err;
         {ok, DeltaTTL = {delta, _}} ->
-            Callback = fun(API, ChainState) -> API:oracle_extend(<<Oracle:256>>, to_sign(Sign0), TTL, ChainState) end,
-            DynCost = state_gas_cost(oracle_extension, DeltaTTL),
-            ?TEST_LOG("~s computed gas cost ~p from relative TTL ~p", [?FUNCTION_NAME, DynCost, DeltaTTL]),
-            {ok, DynCost, fun() -> cast_chain(Callback, State) end}
+            {ok, Tx} = API:oracle_extend_tx(<<Oracle:256>>, TTL, State),
+            Callback = fun(ChainAPI, ChainState) ->
+                               ChainAPI:oracle_extend(Tx, to_sign(Sign0), ChainState)
+                       end,
+            SizeGas = size_gas(Tx),
```

Comment:
This one should not be charged, we do not add anything to the state tree! Though the StateGas makes sense to have.